src/main/java/software/amazon/event/kafkaconnector/EventBridgeWriter.java

```diff
@@ -120,32 +125,47 @@ public EventBridgeWriter(EventBridgeAsyncClient ebClient, EventBridgeSinkConfig
    * @param records
    * @return list of all records with additional status information
    */
-  public List<EventBridgeWriterRecord> putItems(List<EventBridgeWriterRecord> records) {
+  public EventBridgeWriterResult putItems(List<SinkRecord> records) {
+
+    var putEventsEntries = new ArrayList<Pair<PutEventsRequestEntry, SinkRecord>>(records.size());
     try {
-      var putEventsEntries =
+      var partition =
           records.stream()
               .map(this::createPutEventsEntry)
-              .flatMap(Optional::stream)
-              .collect(Collectors.toList());
+              .collect(partitioningBy(it -> it.right().isPresent()));
+
+      var failedMappedSinkRecords =
```

Comment:
After the parition I would propose that we already get the two lists ready and checked. Otherwise we will do this .get / left / right and Optional check - on every occurence later in the code.

```
var partition = //get map of both lists
var success = parition.get(true);
var errors = partitiont.get(false);
```

Reply:
Agree.